prediction_market_agent/tools/betting_strategies.py

```diff
@@ -0,0 +1,215 @@
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+    ```
+
+    x: Number of tokens in the selected outcome pool
+    y: Number of tokens in the other outcome pool
+    p: Probability of winning
+    c: Confidence
+    b: Bankroll
+    f: Fee fraction
+    """
+    if b == 0:
+        return 0
+    numerator = (
+        -4 * x**2 * y
+        + b * y**2 * p * c * f
+        + 2 * b * x * y * p * c * f
+        + b * x**2 * p * c * f
+        - 2 * b * y**2 * f
+        - 2 * b * x * y * f
+        + (
+            (
+                4 * x**2 * y
+                - b * y**2 * p * c * f
+                - 2 * b * x * y * p * c * f
+                - b * x**2 * p * c * f
+                + 2 * b * y**2 * f
+                + 2 * b * x * y * f
+            )
+            ** 2
+            - (
+                4
+                * (x**2 * f - y**2 * f)
+                * (
+                    -4 * b * x * y**2 * p * c
+                    - 4 * b * x**2 * y * p * c
+                    + 4 * b * x * y**2
+                )
+            )
+        )
+        ** (1 / 2)
+    )
+    denominator = 2 * (x**2 * f - y**2 * f)
+    if denominator == 0:
+        return 0
+    kelly_bet_amount = numerator / denominator
+    return int(kelly_bet_amount)
+
+
+def get_kelly_criterion_bet(
+    market_address: str,
+    estimated_p_yes: float,
+    max_bet: xDai,
+) -> t.Tuple[xDai, OutcomeIndex]:
+    market: Market = get_market(market_address)
+    if len(market.outcomeTokenAmounts) != 2:
+        raise ValueError("Only binary markets are supported.")
+
+    current_p_yes = market.outcomeTokenMarginalPrices[0]
+    outcome_index = 0 if estimated_p_yes > current_p_yes else 1
+    estimated_p_win = estimated_p_yes if outcome_index == 0 else 1 - estimated_p_yes
+
+    kelly_bet_wei = _get_kelly_criterion_bet(
+        x=market.outcomeTokenAmounts[outcome_index],
+        y=market.outcomeTokenAmounts[1 - outcome_index],
+        p=estimated_p_win,
+        c=1,  # confidence
+        b=xdai_to_wei(max_bet),  # bankroll, or max bet, in Wei
+        f=(xdai_to_wei(1) - market.fee) / xdai_to_wei(1),  # fee fraction
+    )
+    return wei_to_xdai(kelly_bet_wei), outcome_index
+
+
+if __name__ == "__main__":
+    market_address = "0xa3e47bb771074b33f2e279b9801341e9e0c9c6d7"
+    est_p_yes = 0.1
+    mov_bet = get_market_moving_bet(
+        market_address=market_address,
+        target_p_yes=est_p_yes,
+        verbose=True,
+    )
+    kelly_bet = get_kelly_criterion_bet(
+        market_address=market_address,
+        estimated_p_yes=est_p_yes,
+        max_bet=xDai(10),  # This significantly changes the outcome
+    )
+
+    def get_outcome_str(outcome_index: OutcomeIndex) -> str:
+        return "'Yes'" if outcome_index == 0 else "'No'"
```

Comment:
What about adding

```
def get_outcome_str(self, outcome_index: int) -> str:
        return self.outcomes[outcome_index]
```

to the `Market` class, so that it's generally usable for any markets?